Wrappers/Python/ccpi/web_viewer/trame_viewer.py

```diff
@@ -0,0 +1,799 @@
+        self.slice_window_range_slider = self.construct_slice_window_range_slider()
+        self.slice_window_slider = self.construct_slice_window_slider()
+        self.slice_level_slider = self.construct_slice_level_slider()
+
+        # Reconstruct the drawer and push it
+        self.construct_drawer_layout()
+        update_layout(self.layout)
+
+    def change_slice_visibility(self, visbility):
+        if visbility:
+            self.cil_viewer.imageSlice.VisibilityOn()
+            self.disable_2d = False
+        else:
+            self.cil_viewer.imageSlice.VisibilityOff()
+            self.disable_2d = True
+        self.create_drawer_ui_elements()
+        update_layout(self.layout)
+        self.cil_viewer.updatePipeline()
+
+    def change_volume_visibility(self, visibility):
+        if not self.cil_viewer.volume_render_initialised:
+            self.cil_viewer.installVolumeRenderActorPipeline()
+
+        if visibility:
+            self.cil_viewer.volume.VisibilityOn()
+            self.cil_viewer.light.SwitchOn()
+            self.disable_3d = False
+        else:
+            self.cil_viewer.volume.VisibilityOff()
+            self.cil_viewer.light.SwitchOff()
+            self.disable_3d = True
+        self.create_drawer_ui_elements()
+        update_layout(self.layout)
+        self.cil_viewer.updatePipeline()
+
+    def change_background_colour(self, colour):
+        if colour == "cil_viewer_blue":
+            color_data = (.1, .2, .4)
+        else:
+            color_data = getattr(colors, colour.lower())
+        self.cil_viewer.ren.SetBackground(color_data)
+        self.cil_viewer.updatePipeline()
+
+    def change_clipping(self, clipping_on):
+        app = vuetify.get_app_instance()
+        app.set(key="slice_visibility", value=False)
+        if hasattr(self.cil_viewer, "planew"):
+            self.cil_viewer.planew.SetEnabled(clipping_on)
+            self.cil_viewer.ren.Render()
+        else:
+            planew = self.cil_viewer.style.CreateClippingPlane()
+            planew.On()
+        self.cil_viewer.updatePipeline()
+
+    def remove_clipping_plane(self):
+        if hasattr(self.cil_viewer, "planew"):
+            app = vuetify.get_app_instance()
+            app.set(key="toggle_clipping", value=False)
+            
+            self.cil_viewer.volume.GetMapper().RemoveAllClippingPlanes()
+
+            # Now remove planew from the cil_viewer
+            del self.cil_viewer.planew
+
+            self.cil_viewer.getRenderer().Render()
+            self.cil_viewer.updatePipeline()
+
+    def show_slice_histogram(self, show_histogram):
+        self.cil_viewer.updateSliceHistogram()
+        if show_histogram:
+            self.cil_viewer.histogramPlotActor.VisibilityOn()
+        else:
+            self.cil_viewer.histogramPlotActor.VisibilityOff()
+        self.html_view.update()
+
+    def change_slice_number(self, slice_number):
+        self.cil_viewer.updateSliceHistogram()
```

Comment:
We should move this call to updateSliceHistogram into the CILViewer's updatePipeline() - so then it occurs within the viewer so if we made it as an interaction in the viewer (without trame) then it would be automatically updated when the slice number is changed still.